### PR TITLE
on failure to get user email, use default mail information and not fa…

### PR DESCRIPTION
**Pull request by cviecco**
…il there

### CODE REVIEW

File: cmd/keymasterd/idp_oidc.go [hunk 1]
```diff
@@ -308,7 +308,7 @@ func (state *RuntimeState) idpOpenIDCTokenHandler(w http.ResponseWriter, r *http
 		state.writeFailureResponse(w, r, http.StatusBadRequest, "bad code")
 		return
 	}
-	logger.Printf("tok=%+v", tok)
+	logger.Printf("token request tok=%+v", tok)
```
Comment (rgooch) [inline cmd/keymasterd/idp_oidc.go]: Please change to a Debugf.

File: cmd/keymasterd/idp_oidc.go [hunk 2]
```diff
@@ -324,21 +324,31 @@ func (state *RuntimeState) idpOpenIDCTokenHandler(w http.ResponseWriter, r *http
 	//formClientID := r.Form.Get("clientID")
 	logger.Printf("%+v", r)
 
+	unescapeAuthCredentials := true
 	clientID, pass, ok := r.BasicAuth()
 	if !ok {
-		state.writeFailureResponse(w, r, http.StatusUnauthorized, "")
-		return
+		logger.Printf("warn: basic auth Missing")
```
Comment (rgooch) [inline cmd/keymasterd/idp_oidc.go]: Debugf, please.

File: cmd/keymasterd/idp_oidc.go [hunk 3]
```diff
@@ -324,21 +324,31 @@ func (state *RuntimeState) idpOpenIDCTokenHandler(w http.ResponseWriter, r *http
...
+		unescapedClientID, err := url.QueryUnescape(clientID)
+		if err == nil {
+			clientID = unescapedClientID
+		}
+		unescapedPass, err := url.QueryUnescape(pass)
+		if err == nil {
+			pass = unescapedPass
+		}
 	}
 	//logger.Debugf(3, "username=%s, pass=%s", clientID, pass)
```
Comment (rgooch) [inline cmd/keymasterd/idp_oidc.go]: Please either uncomment or delete. There should be no commented out Debugf calls.

File: cmd/keymasterd/idp_oidc.go [hunk 4]
```diff
@@ -534,11 +544,15 @@ func (state *RuntimeState) idpOpenIDCUserinfoHandler(w http.ResponseWriter, r *h
...
+	if len(state.Config.OpenIDConnectIDP.DefaultEmailDomain) > 3 {
+		defaultEmailDomain = state.Config.OpenIDConnectIDP.DefaultEmailDomain
+	}
+	email := fmt.Sprintf("%s@%s", parsedAccessToken.Username, defaultEmailDomain)
 	userAttributeMap, err := state.getUserAttributes(parsedAccessToken.Username, []string{"mail"})
 	if err != nil {
-		logger.Printf("failed to get user attributes %s", err)
-		state.writeFailureResponse(w, r, http.StatusInternalServerError, "")
+		logger.Printf("warn: failed to get user attributes for %s, %s", parsedAccessToken.Username, err)
+		//state.writeFailureResponse(w, r, http.StatusInternalServerError, "")
```
Comment (rgooch) [inline cmd/keymasterd/idp_oidc.go]: Please remove dead (commented-out) code.

**Comment by rgooch** [comment]
Build is failing on MacOS.

**Comment by cviecco** [comment]
With golang 1.9.4 the a darwin dependecy breaks (flynn/hid). Already wrote  Pull request to upstream (https://github.com/flynn/hid/pull/4)

**Pull request merged**